pkg/scheduler/schedule_one_test.go

```diff
@@ -2483,7 +2483,7 @@ func TestSchedulerSchedulePod(t *testing.T) {
 				test.registerPlugins, "",
 				frameworkruntime.WithSnapshotSharedLister(snapshot),
 				frameworkruntime.WithInformerFactory(informerFactory),
-				frameworkruntime.WithPodNominator(internalqueue.NewPodNominator(informerFactory.Core().V1().Pods().Lister())),
+				frameworkruntime.WithPodNominator(internalqueue.NewTestPodNominator(informerFactory.Core().V1().Pods().Lister())),
```

Comment:
Why do we have to use a fake PodNominator during tests actually?
Can we just do without `WithPodNominator` option?

Reply:
It's not possible to remove `WithPodNominator` from tests. There are two use cases of it:
- Using `NominatedPodsForNode` in `RunFilterPluginsWithNominatedPods` ([ref](https://github.com/kubernetes/kubernetes/blob/2b0153132eab82e346c50d4919a8c8adb062c8d3/pkg/scheduler/framework/runtime/framework.go#L1028)). It is called in default_preemption and schedule_one during the tests.
- Using `AddNominatedPod` directly in tests.

Reply:
No problem, It's always better to ask